src/Symfony/Component/HttpKernel/HttpCache/Store.php

```diff
@@ -127,7 +127,7 @@ public function lookup(Request $request)
         // find a cached entry that matches the request.
         $match = null;
         foreach ($entries as $entry) {
-            if ($this->requestsMatch(isset($entry[1]['vary'][0]) ? $entry[1]['vary'][0] : '', $request->headers->all(), $entry[0])) {
+            if ($this->requestsMatch(isset($entry[1]['vary'][0]) ? implode(', ', $entry[1]['vary']) : '', $request->headers->all(), $entry[0])) {
```

Comment:
There's one more place in [this class where only the first vary header is used](https://github.com/jakzal/symfony/blob/bugfix/http-kernel-12118-vary-headers-passed-as-array/src/Symfony/Component/HttpKernel/HttpCache/Store.php#L192). It doesn't seem to be an issue in there though. At least I failed to introduce a failing test justifying a change (so far).